src/test/routes.test.js

```diff
@@ -18,78 +58,181 @@ beforeAll(async () => init.then((res) => {
-        if (res.statusCode !== 200) throw new Error('Bad status code ', res.statusCode);
-        if (res.body.success === true) throw new Error('Empty request body returned successfully', res.statusCode);
+async function testDelay(routeResponseBusData) {
+    const { stopID, tripIdentifuers } = routeResponseBusData;
+    await Promise.all(tripIdentifuers.map(async (tripID): Promise<number> => {
+        const delayReq = `${delay}?stopID=${stopID}&tripID=${tripID}`;
+        await request(server).get(delayReq).expect((delayRes) => {
+            checkDelayResponseValid(delayRes);
+            delays += (delayRes.body.data === null) ? 0 : 1;
+        });
     }));
+    return false;
+}
 
-    // const route1 = `?start=42.444759,-76.484183&end=42.442503,-76.485845&time=${epochTime}&arriveBy=false&destinationName="Schwartz"`;
-    // test(`now: ${route}${route1}`, () => request(server).get(route + route1).expect((res) => {
-    //     checkRouteValid(res);
-    // }));
+async function testTracking(routeResponseBusDataArr) {
+    await request(server).post(tracking, { data: routeResponseBusDataArr }).expect((res) => {
+        checkResponseValid(res);
+    });
+    return false;
+}
 
-    const afternoonYesterday = moment().endOf('day').subtract(36, 'hours').unix();
-    const route2 = `?start=42.444759,-76.484183&end=42.442503,-76.485845&time=${afternoonYesterday}&arriveBy=false&destinationName="Schwartz"`;
-    test(`afternoon yesterday: ${route}${route2}`, () => request(server).get(route + route2).expect((res) => {
-        checkRouteValid(res);
+describe('delay endpoint', () => {
+    test(`${delay}empty params`, () => request(server).get(delay).expect((res) => {
+        checkDelayResponseValid(res);
     }));
 
-    const afternoonToday = moment().endOf('day').subtract(12, 'hours').unix();
-    const route3 = `?start=42.444759,-76.484183&end=42.442503,-76.485845&time=${afternoonToday}&arriveBy=false&destinationName="Schwartz"`;
-    test(`afternoon today: ${route}${route3}`, () => request(server).get(route + route3).expect((res) => {
-        checkRouteValid(res);
-    }));
+    // not needed, use route data from before
+    /*
+    test(`${delay}route1noon`, async () => {
+        await request(server).get(route + route1noon).then(async (res) => {
+            const busData = checkRouteValid(res, true, true);
+            const { stopID, tripIdentifuers } = busData;
+            console.log(busData);
+
+            await Promise.all(tripIdentifuers.map(async (tripID): Promise<number> => {
+                const delayReq = `${delay}?stopID=${stopID}&tripID=${tripID}`;
+                console.log(delayReq);
+                await request(server).get(delayReq).expect((delayRes) => {
+                    console.log(delayRes.body);
+                    checkDelayResponseValid(delayRes);
+                });
+            }));
+        });
+    });
+    */
 
     // TODO
```

Comment:
delete the TODO?

Reply:
It's just to signal we need to write more tests, I'm going to leave it.